### PR TITLE
[Merged by Bors] - feat(Manifold/IntegralCurve/Transform): pointwise notation and translation lemmas for subtraction

### PR DESCRIPTION
**Pull request by winstonyin**
We restate the translation lemmas for integral curves using the `Pointwise` API and add translation lemmas for subtraction for convenience.

This is just split out from #9013.

The `Pointwise` API allows us to use lemmas like `Metric.vadd_ball` and more convenient rewriting of the `dt` term (rather than rewriting it inside the set builder notation).

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Geometry/Manifold/IntegralCurve/Transform.lean [hunk 1]
```diff
@@ -33,30 +33,40 @@ variable
...
   refine ⟨fun hγ ↦ hγ.comp_add _, fun hγ ↦ ?_⟩
   convert hγ.comp_add (-dt)
   · ext t
     simp only [Function.comp_apply, neg_add_cancel_right]
-  · simp only [mem_setOf_eq, neg_add_cancel_right, setOf_mem_eq]
+  · simp only [neg_neg, vadd_neg_vadd]
+
+lemma isIntegralCurveOn_comp_sub {dt : ℝ} :
+    IsIntegralCurveOn γ v s ↔ IsIntegralCurveOn (γ ∘ (· - dt)) v (dt +ᵥ s) := by
+  nth_rw 2 [← neg_neg dt]
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/IntegralCurve/Transform.lean:56]: I wonder: can you prove this using `isIntegralCurveOn_comp_add`? For the other analogues, some `simpa using ..._comp_add` works.

**Review by grunweg — COMMENTED** [review]
Basically looks good to me - I just wonder if one proof can be simplified.

**Comment by kim-em** [comment]
bors d+

**Pull request closed without merging**